server/middleware/versions.coffee

```diff
@@ -89,7 +94,11 @@ module.exports =
 
     doc.set('parent', latest._id)
 
-    doc = yield doc.save()
+    try
+      doc = yield doc.save()
+    catch e
+      yield latest.update({$set: _.pick(latest.toObject(), 'version', 'index', 'slug')})
+      throw e
```

Comment:
`latest` is changed a few lines before, [here](https://github.com/codecombat/codecombat/pull/3879/files#diff-7902f1d8909a70a930595eef60a6e388R72). It's not very clear that it's happening, though. I'll add comments, and actually adjust the logic a bit to be clearer.